Remove gin

dukex opened the pull request:
Just the first step to remove gin, now we should add logging, tracing, maybe a cache to improve the server. 

We have good resource and example here https://gist.github.com/enricofoltran/10b4a980cd07cb02836f70a4ab3e72d7

Closes #20

dukex committed:
remove gin from handler package

dukex committed:
remove gin from config and from server

dukex committed:
update deps

dukex committed:
refactor handler

ghost commented:
[**DeepCode Report (#0413f8)**](https://www.deepcode.ai/app/gh/leogregianin/brcep/9251ce1ec64f06d5270e34424ca05531d8d65c5c/leogregianin/brcep/0413f80f3dc27a7872b2451142cb692496bc4a32/_/%2F/code/)
---
DeepCode analyzed this pull request.
There are no new issues.

ghophp commented on handler/handler.go:
```diff
@@ -12,23 +13,34 @@ type CepHandler struct {
+}
+
+func (h *CepHandler) Handle(w http.ResponseWriter, r *http.Request) {
+	cep := strings.Split(r.URL.Path, "/")[1]
```
this has to be more defensive, it can trigger a panic that can make the server stop.. since we don't have an easy way to map path parameters from the URL, what about introducing a method like:

```go
cep, err := h.getCepFromUrl(r.URL)
if err != nil {
   renderJSON(w, http.StatusBadRequest, &resonseError{"Invalid cep provided"})
}

var invalidCep = errors.New("invalid cep provided")

func (h *CepHandler) getCepFromUrl(url url.URL) (string, error) {
   var path = r.URL.Path
   if len(path) <= 0 {
        return "", invalidCep
   }

   var pathParts strings.Split(path, "/")
   if len(pathParts) < 2 {
        return "", invalidCep
   }

   return pathParts[1]
}
```

and then please extend the tests to match the scenarios 👍

ghophp commented on handler/handler.go:
```diff
@@ -12,23 +13,34 @@ type CepHandler struct {
 	preferredApi, ok := h.CepApis[h.PreferredApi]
 	if !ok {
-		ctx.JSON(http.StatusInternalServerError, gin.H{"error": "preferred api not available"})
+		renderJSON(w, http.StatusInternalServerError, &responseError{Error: "preferred api not available"})
```
you can omit the `Error:` when instantiating the struct here, so `&responseError{"preferred api not available"}`

ghophp reviewed (changes requested):
thanks for the contribution @dukex very nice removal, I also thought it was completely unnecessary to bring gin as a dependency for such a small case.. I made just a small comment about be more defensive when extracting the cep from the URL, besides that LGTM 👍

dukex committed:
apply PR suggestions

dukex committed:
remove empty line

dukex committed:
fix merge errors

dukex commented:
Thanks @ghophp, can you review it again?

ghophp committed:
small changes to increase redeability

in order to make it easier to follow where the invalidCep variable comes from, better to return it directly on the method

ghophp committed:
small fix due to invalid syntax

ghophp committed:
undefined variable
